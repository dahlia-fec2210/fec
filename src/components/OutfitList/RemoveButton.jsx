/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';

function RemoveButton({ removeFromOutfit, clothingPiece }) {
  return (
    <div className="related-stack" onClick={removeFromOutfit}>
      <div className="fa-stack" style={{ verticalAlign: 'top' }}>
        <i className="related-circle fa-solid fa-regular fa-circle fa-stack-2x" />
        <i id={`outfit-list-remove-button:${clothingPiece}`} className="related-star fa-solid fa-x fa-stack-1x" />
      </div>
    </div>
  );
}

export default RemoveButton;
