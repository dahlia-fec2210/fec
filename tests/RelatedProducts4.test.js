/* eslint-disable no-undef */
/* eslint-disable import/extensions */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import RemoveButton from '../src/components/OutfitList/RemoveButton.jsx';

test('renders remove button on outfit list with product 37315', async () => {
  const tree = renderer
    .create(<RemoveButton clothingPiece={37315} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
