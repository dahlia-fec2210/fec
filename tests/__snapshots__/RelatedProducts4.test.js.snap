// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders remove button on outfit list with product 37315 1`] = `
<div
  className="related-stack"
>
  <div
    className="fa-stack"
    style={
      {
        "verticalAlign": "top",
      }
    }
  >
    <i
      className="related-circle fa-solid fa-regular fa-circle fa-stack-2x"
    />
    <i
      className="related-star fa-solid fa-x fa-stack-1x"
      id="outfit-list-remove-button:37315"
    />
  </div>
</div>
`;
